content/blog/2020-07-31-spreadsheet-workflows-using-r/index.Rmarkdown

```diff
@@ -0,0 +1,185 @@
+Before we start exploring this data we'll need to join in the relevant data so we can conduct some fun analyses. 
+```{r, include= FALSE}
+superheroes <- read_excel("data/superheroes.xlsx", sheet = "heroes")
+identities <- read_excel("data/superheroes.xlsx", sheet = "alter_egos")
+```
+
+Let's take a look at our initial sheet on superheores
+```{r}
+superheroes
+```
+
+And now another sheet lisitng the alter ego of each character
+```{r}
+identities
+```
+
+We see in each sheet there is a hero column allowing us to bring in data from one sheet to the other.   
+Using `left_join()` from the dplyr package, nearest equivalent to the vlookup, we can bring in the data from the identities sheet into our initial superheroes data  
+
+To do this we'll call `left_join()` on superheores, putting the identities dataset as our first argument and specify our common varible with the `by` argument 
+```{r}
+superhero_identities <- superheroes %>%
+  left_join(identities, by = "hero")
+
+superhero_identities
+```
+
+Great, we've run our join no problem. Of course however we'll run into instances in which we'll need to pull in data from multiple sheets and the superheroes dataset is no exception. 
+We've been informed there are an additional two sheets that have recently been added, one containing information on the main power or skill of each characrer...
+```{r,include=FALSE}
+powers <- readxl::read_excel("data/superheroes.xlsx", sheet = "powers")
+```
+
+```{r}
+powers
+```
+
+And the other containing the occupations of the each heroes alter-ego. 
+```{r}
+occupation <- readxl::read_excel("data/superheroes.xlsx", sheet = "occupation")
+```
+
+```{r}
+occupation
+```
+
+We see that the powers sheet has a hero column  allowing for another simple left join, however the occupation sheet doesn't have a common column by specific name but we do recognize that the information in the name column is the same for the alter-ego column from our identities spreadsheet.
+We can take advantage of this relationship to bring this data in as shown in the code below.  
+```{r}
+heroes_full <- superhero_identities %>%
+  left_join(powers, by = "hero") %>%
+  left_join(occupation, by = c("alter_ego" = "name"))
+
+```
+Here we conduct a simple left join bringing in the data from the powers dataset, and then another left joing by alter ego to bring in the data from the occupation spreadsheet. 
+In the `by` argument we can specify that the column `alter_ego` should be matched onto the `name` column allowing for the join to complete. 
+
+We've expanded our dataset quite a bit. Let's take an alternative apprach to viewing it using the `glimpse()` function from dplyr
+```{r}
+glimpse(heroes_full)
+```
+There we have it, a fully combined dataset ready for analysis!  Hmmm, now that we think about it, let's hope we aren't sharing this data with any supervillains.  We may be an ace at working with spreadsheet data but we're also an ethical data analyst!
```

Comment:
Overall, I have the same comment for the join work as for pivot table and nested `=IF()`. I'd love to see a literal juxtaposition of "The Excel (or spreadsheets) Way" vs. "The R Way".

Reply:
Completely agree - in the article now we have comparisons between tidyverse code and excel examples